### PR TITLE
update datafusion to 42

### PR DESCRIPTION
**Pull request by davidhewitt**
Waiting on datafusion 42 release (any day now).

### CODE REVIEW

File: tests/main.rs [hunk 1]
```diff
@@ -944,12 +944,12 @@ async fn test_arrow_nested_double_columns() {
...
-    let sql = r#"select '{"a": "b"}'->>'a'='b' as v"#;
-    let err = run_query(sql).await.unwrap_err();
-    assert_eq!(err.to_string(), "Error during planning: Unexpected argument type to 'json_as_text' at position 2, expected string or int, got Boolean.")
-}
+// #[tokio::test]
+// async fn test_lexical_precedence_wrong() {
+//     let sql = r#"select '{"a": "b"}'->>'a'='b' as v"#;
+//     let err = run_query(sql).await.unwrap_err();
+//     assert_eq!(err.to_string(), "Error during planning: Unexpected argument type to 'json_as_text' at position 2, expected string or int, got Boolean.")
+// }
```
Comment (davidhewitt) [inline tests/main.rs:952]: @samuelcolvin I assume this test was documenting broken parse precedence and we can now just remove it?

Comment (davidhewitt) [reply tests/main.rs:952]: (Or update it to test the successful case?)

**Comment by hengfeiyang** [comment]
Datafusion already release 42

File: tests/main.rs (hunk 1 above)
Comment (samuelcolvin) [reply tests/main.rs:952]: yes, I'll fix.

**Comment by samuelcolvin** [comment]
superceded by #47, it was easier to create a new PR due to conflicts. We can close this once we're no longer using a sha from  this branch in logfire.

**Pull request closed without merging**